src/main/java/io/github/treesitter/jtreesitter/QueryCursor.java

```diff
@@ -293,7 +293,8 @@ public static class Options {
         private final @Nullable BiPredicate<QueryPredicate, QueryMatch> predicateCallback;
 
         /**
-         * @param progressCallback Progress handler.
+         * @param progressCallback Progress handler. Return {@code true} to cancel query execution,
+         *                         {@code false} to continue query execution.
```

Comment:
Same as the comment in `Parser`; I hope I understood the logic in https://github.com/tree-sitter/tree-sitter/blob/37a9ecd5b7d52786d0673611f0b8fc54292b1ac1/lib/src/query.c#L3631 correctly, and `true` does actually cancel query execution.